SingularityUI/app/components/machines/SlaveUsage.jsx

```diff
@@ -0,0 +1,211 @@
+import React, { PropTypes } from 'react';
+import { Dropdown, Nav, NavItem } from 'react-bootstrap';
+import { Glyphicon } from 'react-bootstrap';
+import Utils from '../../utils';
+import { connect } from 'react-redux';
+import rootComponent from '../../rootComponent';
+import { FetchSlaveUsage, FetchSlaves } from '../../actions/api/slaves';
+import CopyToClipboard from 'react-copy-to-clipboard';
+
+const SlaveUsage = (props) => {
+  const navigation = (
+    // TODO: fix justified
+    <Nav bsStyle='pills' activeKey={2} justified={true}>
+      <NavItem eventKey={1} title='aggregate'> Aggregate </NavItem>
+      <NavItem eventKey={2} title='heatmap'> Heatmap </NavItem>
+    </Nav>
+  );
+
+  const drawSlaves = () => {    
+    var grid = props.slaveUsage.map((slave, index) => {
+      return drawSlave(slave, index);
+    });
+    return grid;
+  };
+
+  const drawSlave = (slave, index) => {
+    var criticalGlyph = 'glyphicon glyphicon-remove-sign';
+    var warningGlyph = 'glyphicon glyphicon-minus-sign';
+    var okGlyph = 'glyphicon glyphicon-ok-sign';
+    var criticalStyle = 'danger';
+    var warningStyle = 'warning';
+    var okStyle = 'success';
+
+    if (isStatCritical(slave, props.cpusUsedStat) || isStatCritical(slave, props.memoryBytesUsedStat) || isStatCritical(slave, props.numTasksStat)) {
```

Comment:
we should probably catch the exceptions you are throwing in the case where we can't properly find the stats. It'd be better to show less stats (excluding the ones that threw an exception), rather than throwing js errors all the way up